docs/hugo/content/design/ADR-2023-09-Skipping-Properties/_index.md

```diff
@@ -0,0 +1,418 @@
+[v4.Person]->[v4.Address|Label multilineString|Properties PropertyBag]
+
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress v5.Address|Properties PropertyBag]
+[v5.Person]->[v5.Address|Street string; Suburb string; City string; Country string|Properties PropertyBag]
+-->
+
+{{< figure src="./option-1-class.png" >}}
+
+The shape of `Address` when reintroduced will always match the shape of `Address` when removed, so we can always serialize the same shape into or outof the property bag, ensuring any `Address` has the a consistent shape.
+
+Pros:
+* Works the way existing conversions do.
+* We have an existing contract for augmenting conversions with custom code, allowing us to handcode the conversion between **v4** and **v5** to handle the mismatch in the shapes of `Address`.
+* Addresses the central inconsistency problem, ensuring we always serialize the same shape into intermediate property bags
+
+Cons:
+* Changes the serialization format for existing **v4** resources. To avoid a breaking change, we'd need to code generate conversions for several edge cases, including:
+  * Conversion of a _new style_ **v4** `Person` to **v3** person, handling both the case where the address is in the property bag, and the case where it is a property reference.
+  * Conversion of a _new style_ **v4** `Person` to **v5** person, handling both the case where the address is in the property bag (requiring conversion from the **v3** shape), and the case where it is a property reference (requiring conversion from the **v4** shape).
+* Changes the object model of released resources, a breaking change for code consumers of the object model.
+* When there are multi-generational skips, other questions arise: Do we introduce the new type into every intermediate version, only the last, or both first and last? Each approach has advantages and drawbacks.
+
+### Option 2: Back conversion by new version
+
+Ensure that complex types stored in property bags have a consistent shape by introducing a dedicated type for that purpose. Regardless of whether the property bag is populated by an earlier version (e.g. a **v3** `Address`) or a later version (e.g. a **v5** `Address`), the shape of the object stored property bag will be the same (**v4**).
+
+<!-- yuml.me
+
+[v3.Person|FullName string;FamilyName string; KnownAs string; ResidentialAddress v3.Address|Properties PropertyBag]
+[v3.Person]->[v3.Address|Label multilineString|Properties PropertyBag]
+
+[v4.Person|FullName string;FamilyName string; KnownAs string|Properties PropertyBag]
+[v4.Person]-.-[v4.Address {bg:cornsilk}|Label multilineString|Properties PropertyBag]
+
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress v5.Address|Properties PropertyBag]
+[v5.Person]->[v5.Address|Street string; Suburb string; City string; Country string|Properties PropertyBag]
+
+-->
+
+{{< figure src="./option-2-class.png" >}}
```

Comment:
You inconsistently call the `v4` address `compat.Address` and `v4.Address` in the two option-2 diagrams.

I _think_ you mean to call it `v4.Address` in this (new) option 2?